auth_core/token_data.py

```diff
@@ -0,0 +1,10 @@
+import uuid
+
+from pydantic import BaseModel
+
+
+class TokenData(BaseModel):
+    auth_token: str
+    identity_id: uuid.UUID
+    provider_token: str | None
+    provider_refresh_token: str | None
```

Comment:
`|` for `Union` typing is only available since [Python 3.10](https://peps.python.org/pep-0604/), and looks like [FastAPI supports as low as 3.8](https://github.com/fastapi/fastapi/blob/b0215699137c41a96c57f650640cb9db1c25314a/pyproject.toml#L10). Maybe we should align with that.